feat: add sprig to declare default

edouard-lopez opened the pull request:
- feat: add Sprig to templates for additional templating capabilities
- test: add config.Get()
- refactor(template): extract parseJobTemplate to test it

vsukhin commented on internal/template/parser.go:
```diff
@@ -0,0 +1,73 @@
+
+	"github.com/kubeshop/testkube/internal/config"
+	"github.com/kubeshop/testkube/pkg/log"
+	"github.com/pkg/errors"
```
fix import

vsukhin commented on internal/template/parser.go:
```diff
@@ -0,0 +1,73 @@
+	fmt.Printf("filename: [%v]\n", filename)
+	fmt.Printf("filepath [%v]\n", filepath.Join(configDir, filename))
+	pwd, _ := os.Getwd()
+	fmt.Printf("pwd: [%v]\n", pwd)
```
remove, when finish testing

edouard-lopez commented:
We introduced a `utils.NewTemplate()` to inject the functions from [`slim-sprig`](https://github.com/go-task/slim-sprig)

```diff
-	t, err := template.New("message").Parse(s.messageTemplate)
+	t, err := utils.NewTemplate("message").Parse(s.messageTemplate)
```

Thanks to @adrien-f for the heavy load :pray: 

@vsukhin As I was exploring, I extracted some methods from _cmd/api-server/main.go_ to [internal/template/parser.go](https://github.com/kubeshop/testkube/pull/4339/files#diff-3fe310a846aa857b279ac5b5b815424015ffe7fc75a1aec2b3e5d152a705bbb5) 
 Could you check that extracting the methods  `LoadConfigFromStringOrFile`, `ParseJobTemplate`, `IsBase64Encoded` has no further impact?

vsukhin commented:
thank you, guys @edouard-lopez @adrien-f Will check it on Monday morning

vsukhin reviewed (approved):
looks good. @nicufk (Nico developed above methods) ?

nicufk reviewed (approved):
LGTM. Nice Job!
